### ISSUE TITLE
Caps and Join style don't work for image backend

### ISSUE THREAD

**Issue by GoogleCodeExporter** [llgcode/draw2d#27]
```
It appears that MiterJoin doesn't do anything at all. From reading the source 
code, if the Join style is not RoundJoin or BevelJoin, then the Joiner returned 
is the RoundJoiner. Although in my case, RoundJoin doesn't seem to work either 
(I always get a bevel).
```

Original issue reported on code.google.com by `ke...@sb.org` on 27 Oct 2013 at 10:22

**Comment by stanim** [comment]
I adapted the title as caps and join styles do work for the pdf backend.

**Comment by dax0** [comment]
Is there any update on this issue?

**Comment by fogleman** [comment]
@llgcode What's needs to happen to fix this? I've been digging through the code to try to figure it out.

I don't quite understand why Stroke() in ftgc.go doesn't use AddStroke in the ft Rasterizer. There are a lot of abstractions going on.

**Comment by llgcode** [comment]
Yes you found a limitation of AddStroke Rasterizer that don't implements cubic
bezier curve, so this is implemented differently in draw2d.
https://github.com/golang/freetype/blob/master/raster/stroke.go#L403
I could switch to freetype stroker if it implements cubic bezier curve *and
*dasher. Or you could implement it in dr'aw2d stroker here :
https://github.com/llgcode/draw2d/blob/master/draw2dbase/stroker.go#L72
Thanks for your help and sorry for the delay.

-- Laurent.

**Comment by fogleman** [comment]
@llgcode I ended up writing my own graphics package on top of `golang/freetype/raster`. It supports dashed lines and line caps / joins (and other stuff). Have a look:

https://github.com/fogleman/gg

**Comment by llgcode** [comment]
great!!. Seems to be a serious alternative to draw2d. You've made a great
Job. Your project seems to be popular yet in 8 days !
Your implementation seems to be cleaner.
Can we share/use a same interface
https://github.com/llgcode/draw2d/blob/master/gc.go?

-- Laurent.

**Comment by llgcode** [comment]
Libne cap can be drawn on every dash
https://gist.github.com/llgcode/be814ce3a6df7a6f9f94

**Comment by fogleman** [comment]
Nice!
